azurelinuxagent/common/protocol/wire.py

```diff
@@ -314,45 +321,51 @@ def __init__(self, client):
         self.vm_status = None
         self.ext_statuses = {}
         self.client = client
+        self.type = None
+        self.data = None
 
     def set_vm_status(self, vm_status):
         validata_param("vmAgent", vm_status, VMStatus)
         self.vm_status = vm_status
-    
+
     def set_ext_status(self, ext_handler_name, ext_status):
         validata_param("extensionStatus", ext_status, ExtensionStatus)
-        self.ext_statuses[ext_handler_name]= ext_status
-        
+        self.ext_statuses[ext_handler_name] = ext_status
+
     def to_json(self):
         report = vm_status_to_v1(self.vm_status, self.ext_statuses)
         return json.dumps(report)
 
     __storage_version__ = "2014-02-14"
 
     def upload(self, url):
-        #TODO upload extension only if content has changed
+        # TODO upload extension only if content has changed
         logger.verb("Upload status blob")
-        blob_type = self.get_blob_type(url)
-
-        data = self.to_json()
+        upload_successful = False
+        self.type = self.get_blob_type(url)
+        self.data = self.to_json()
```

Comment:
yes, that way the fallback logic in wire.py can simply pass the status_blob instance along to the host plugin instead of type, status and storage version.